File: NudgeApp.Web/ClientApp/src/app/types/NotificationDto.ts
import { NudgeResult } from "./Nudge";
import { TransportationType } from "./TripDto";
import { SkyCoverageType, WindCondition, RoadCondition, Probabilities } from "./ForecastDto";

export interface NotificationDto {
  id: any,
  text: string,
  createdOn: Date,
  nudgeResult: NudgeResult,
  transportationType: TransportationType,
  tripType: TripType,
  distance: number,
  duration: number,
  reafFeelTemperature: number,
  temperature: number,
  wind: number,
  precipitationProbability: number,
  cloudCoveragePercent: number,
  skyCoverage: SkyCoverageType,
  windCondition: WindCondition,
  probability: Probabilities,
  roadCondition: RoadCondition,
  dateTime: Date
}

export enum TripType {
  Walk,
  WithDestinaion
}

File: NudgeApp.Common/Dtos/NotificationDto.cs
using System;
using NudgeApp.Common.Enums;

namespace NudgeApp.Common.Dtos
{
    public class NotificationDto
    {
        public Guid Id { get; set; }
        public string Text { get; set; }
        public DateTime CreatedOn { get; set; }
        public NudgeResult NudgeResult { get; set; }
        public TransportationType TransportationType { get; set; }
        public TripType TripType { get; set; }
        public int Distance { get; set; }
        public int Duration { get; set; }
        public float ReafFeelTemperature { get; set; }
        public float Temperature { get; set; }
        public float Wind { get; set; }
        public int PrecipitationProbability { get; set; }
        public int CloudCoveragePercent { get; set; }
        public SkyCoverageType SkyCoverage { get; set; }
        public WindCondition WindCondition { get; set; }
        public Probabilities Probability { get; set; }
        public RoadCondition RoadCondition { get; set; }
        public DateTime DateTime { get; set; }
    }
}